Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,152 @@
+    args_parser.add_option(show_paths, "Display the path for each matched file.", nullptr, 'p');
+    args_parser.add_option(string_offset_format_sv, "Write offset relative to start of each file in (d)ec, (o)ct, or (h)ex format.", nullptr, 't', "format");
+    args_parser.set_general_help("Write the sequences of printable characters in files or pipes to stdout.");
+    args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    if (minimum_string_length < 1) {
+        warnln("Invalid minimum string length {}", minimum_string_length);
+        return 1;
+    }
+
+    if (string_offset_format_sv == "d") {
+        string_offset_format = StringOffsetFormat::Decimal;
+    } else if (string_offset_format_sv == "o") {
+        string_offset_format = StringOffsetFormat::Octal;
+    } else if (string_offset_format_sv == "h") {
+        string_offset_format = StringOffsetFormat::Hexadecimal;
+    } else {
+        if (!string_offset_format_sv.is_empty()) {
+            warnln("Invalid format {}", string_offset_format_sv);
+            return 1;
+        }
+    }
+
+    bool all_ok = true;
+
+    Vector<int> fds;
+    if (paths.is_empty()) {
+        TRY(fds.try_append(STDIN_FILENO));
+    } else {
+        for (auto const& path : paths) {
+            int fd;
+            if (path == "-") {
+                fd = 0;
+            } else {
+                auto fd_or_error = Core::System::open(path, O_RDONLY);
+                if (fd_or_error.is_error()) {
+                    all_ok = false;
+                    warnln("Failed to open {}: {}", path, fd_or_error.error());
+                    continue;
+                }
+                fd = fd_or_error.release_value();
+            }
+            TRY(fds.try_append(fd));
+        }
+    }
+
+    TRY(Core::System::pledge("stdio"));
+
+    Array<u8, 32768> buffer;
+    Vector<u8> output_characters;
+    for (auto& fd : fds) {
+        bool did_show_path = false;
+        current_fd_position = string_offset_position = 0;
+        for (;;) {
+            auto buffer_span = buffer.span();
+            auto nread = TRY(Core::System::read(fd, buffer_span));
+            if (nread == 0)
+                break;
+            buffer_span = buffer_span.trim(nread);
+            while (!buffer_span.is_empty()) {
+                int processed_characters = 0;
+                for (u8 ch : buffer_span) {
+                    ++processed_characters;
+                    ++current_fd_position;
+                    if (is_ascii_printable(ch) || ch == '\t') {
+                        if (output_characters.is_empty())
+                            string_offset_position = current_fd_position - 1;
+                        output_characters.append(ch);
+                    } else {
+                        if (output_characters.size() < 4)
```

Comment:
Hardcoded minimum string length?

Reply:
Fixed in [a2a06c7](https://github.com/SerenityOS/serenity/pull/16013/commits/a2a06c7317d5818c0198c123b43bbd6694d49000)

Reply:
No, that is a different location.

Reply:
Oof, sorry about that. My editor had that section collapsed for some reason. Fixed in [7b5de73](https://github.com/SerenityOS/serenity/pull/16013/commits/7b5de738dfdd5e5f02f5141bf3592142a251a14e)